cmd/noms/noms_log.go

```diff
@@ -99,8 +100,12 @@ func runLog(args []string) int {
 	defer pgr.Stop()
 
 	for commitBuff := range outChan {
-		io.Copy(pgr.Writer, bytes.NewReader(commitBuff.([]byte)))
+		_, err := io.Copy(pgr.Writer, bytes.NewReader(commitBuff.([]byte)))
+		if err != nil {
+			os.Exit(0)
```

Comment:
There are a shit ton of commits so even when I exit my pager (less) it keep cranking along (I can see it w/ DTrace).

If I just do `break` it will occasionally hang forever. The short of it is that if you do a `datas.httpBatchStore.Get()` after the `Close()` method of that class has been called you'll hang forever because the background task (spawned from `batchReadRequests()`) that processes those requests has exited.

I've chased this down and found a better fix, but I'd suggest that there are still some gaps here. In particular invoking `Close()` should cause outstanding `Get()` calls to fail and return immediately, but there doesn't appear to be a mechanism to achieve this (e.g. outstanding and in-flight requests are not tracked centrally).

I'd strongly suggesting asserting that untoward stuff doesn't happen after `Close()` as it will certainly lead to problems.

New fix coming.